### PR TITLE
Ethers v6 target

### PR DESCRIPTION
**Pull request by zemse**
For trying out this PR in your project:

```
npm i typechain-target-ethers-v6
```

1. You can install the above temporary package and use `ethers-v6` as target in your typechain config (or you can clone this repo and publish yourself).
2. Make sure ethers version is updated in your project.
3. Make sure typescript version is `>=4.7`.
4. Make sure in your tsconfig: moduleResolution is node16 or nodenext. (Known issue for Rollup https://github.com/ezolenko/rollup-plugin-typescript2/issues/437)

### CODE REVIEW

**Comment by zemse** [comment]
`typechain-target-ethers-v6@0.0.2` is published with this fix.

**Comment by stevealexrs** [comment]
How do I provide structs as function input? The struct is declared as 
```ts
export declare namespace ContractName {
  export type SignatureStruct = {
    signer: AddressLike;
    v: BigNumberish;
    r: BytesLike;
    s: BytesLike;
  };
```

**Comment by zemse** [comment]
That should be `import type { ContractName } from 'path'`, notice "type" is outside of the `{}`. Or if that was a typo can you describe more about the error you are facing?
```ts
// full path to the contract typing file, it is not exported from index.ts
import type { ISwapRouter } from '../typechain';
const input: ISwapRouter.ExactInputParamsStruct = {
  path: '0x1234',
  recipient: '0x000',
  deadline: 1,
  amountIn: 1,
  amountOutMinimum: 1,
};
```

File: packages/target-starknet-test/package.json [hunk 1]
```diff
@@ -26,6 +26,6 @@
   },
   "dependencies": {
     "@types/elliptic": "^6.4.14",
-    "starknet": "^3.9.0"
```
Comment (krzkaczor) [inline packages/target-starknet-test/package.json:29]: any specific reason for this change?

Comment (zemse) [reply packages/target-starknet-test/package.json:29]: Did this in [2085673](https://github.com/dethcrypto/TypeChain/pull/803/commits/2085673e067a6b800aad221d83153c733e232af6) temporarily to try to fix CI along with [6ce08b5](https://github.com/dethcrypto/TypeChain/pull/803/commits/6ce08b5b77635a57640d67af06cbc5d8c83378b7) and [eacbe6d](https://github.com/dethcrypto/TypeChain/pull/803/commits/eacbe6d03d9a98dcec8d38b1384d7fc6125b907e). Reverting them.

**Comment by krzkaczor** [comment]
@zemse thank you so much for this work. 

Final fixes are done in: https://github.com/dethcrypto/TypeChain/pull/808

**Pull request closed without merging**